src/EnergyPlus/InputProcessing/CsvParser.cc

```diff
@@ -233,53 +201,49 @@ void CsvParser::parse_header(std::string_view csv, size_t &index, bool &success,
             next_token(csv, index);
         } else {
             header.push_back(parse_value(csv, index));
-            if (!success) return;
         }
     }
 }
 
 void CsvParser::parse_line(std::string_view csv, size_t &index, json &columns)
 {
     Token token;
-    int column_num = 0;
+    size_t column_num = 0;
+    size_t parsed_values = 0;
+    const size_t num_columns = columns.size(); // Csv isn't empty, so we know it's at least 1
+
+    size_t this_cur_line_num = cur_line_num;
+    size_t this_beginning_of_line_index = beginning_of_line_index;
 
     while (true) {
         token = look_ahead(csv, index);
         if (token == Token::LINE_END || token == Token::FILE_END) {
+            if (parsed_values != num_columns) {
+                success = false;
+
+                size_t found_index = csv.find_first_of('\n', this_beginning_of_line_index);
+                std::string line;
+                if (found_index != std::string::npos) {
+                    line = csv.substr(this_beginning_of_line_index, found_index - this_beginning_of_line_index);
+                }
+                errors_.emplace_back(
+                    fmt::format(
+                        "CsvParser - Line {} - Expected {} columns, got {}. Error in following line.", this_cur_line_num, num_columns, parsed_values),
+                    false);
+                errors_.emplace_back(line, true);
+            }
```

Comment:
When we're finished with the line, check that we have the proper amount of values. Otherwise, emplace errors.